src/Components/SingleArticle.jsx

```diff
@@ -17,6 +19,12 @@ function SingleArticle() {
     })
   }, [article_id]);
 
+  const upVote = (articleId) => {
+    setSingleArticle((currArticle) => {
+      return {...currArticle, votes: currArticle.votes + 1}
+    })
+    patchArticle(articleId)
+  }
```

Comment:
Very good. You may want to un-do this optimistic rendering if anything goes wrong with the patch request. You can simply un-do it in a catch block :)